### PR TITLE
1061 fix sn editor tiptap extension bugs

### PR DESCRIPTION
**Pull request by Midnite1212**
Bible verse (Richie)
- [x]  •⁠ ⁠add different chapter Bible verse extension

User notes (Richie)

- [x]  •⁠ ⁠Make user notes full width
- [x]  •⁠ ⁠User notes line break not working
- [x]  •⁠ ⁠Need to login to show SN

### CODE REVIEW

File: ui/src/components/helpers/SermonNotes.js [hunk 1]
```diff
@@ -19,6 +19,35 @@ export const getBiblePassage = async (passage, currentText) => {
     }
     const [chapter, verse] = chapterAndVerse.split(':');
 
+    // handle if its different chapter
+    const isDifferentChapter = chapterAndVerse.split(':').length > 2;
+    if (bible && bible[book] && isDifferentChapter) {
+      const [firstChapterVerse, lastChapterVerse] = chapterAndVerse.split('-');
+      const [firstChapter, firstVerse] = firstChapterVerse.split(':');
+      const [lastChapter, lastVerse] = lastChapterVerse.split(':');
```
Comment (Baeck016) [inline ui/src/components/helpers/SermonNotes.js:27]: So it will work max 2 chapters right?

**Comment by Baeck016** [comment]
I don't think vv. or v. is working

**Comment by Midnite1212** [comment]
> I don't think vv. or v. is working 
> 

Did u input the passage? It only works when the passage verse is inputted.

**Comment by Baeck016** [comment]
> > I don't think vv. or v. is working
> 
> Did u input the passage? It only works when the passage verse is inputted. Cuz it diesnt have book and chapter, so it relies on the passage in the form.

image

File: ui/src/components/helpers/SermonNotes.js (hunk 1 above)
Comment (Midnite1212) [reply ui/src/components/helpers/SermonNotes.js:27]: Nope, it shod work with any length of chapters

**Comment by Midnite1212** [comment]
> > > I don't think vv. or v. is working
> > 
> > Did u input the passage? It only works when the passage verse is inputted. Cuz it diesnt have book and chapter, so it relies on the passage in the form.
> 
> image
> 

I mean the passage in the form above, cuz idt tiptap lets me to take the passage from TEXT: lol

File: ui/src/components/helpers/SermonNotes.js [hunk 2]
```diff
@@ -19,6 +19,35 @@ export const getBiblePassage = async (passage, currentText) => {
...
+      let passageBlock = '';
+      for (let i = Number(firstChapter); i <= lastChapter; i++) {
```
Comment (bhryck) [inline ui/src/components/helpers/SermonNotes.js:29]: perhaps can rename i and j for readability
```suggestion
      for (let currentChapter = Number(firstChapterNum); currentChapter <= Number(lastChapterNum); currentChapter++) {
    if (currentChapter === Number(firstChapterNum)) {
      for (let currentVerse = Number(firstVerseNum); bible[book][currentChapter][currentVerse] !== undefined; currentVerse++) {
        passageBlock += `${currentVerse} ${bible[book][currentChapter][currentVerse]} `;
      }
      passageBlock += '<br />';
    } else if (currentChapter === Number(lastChapterNum)) {
      passageBlock += '<br />';
      for (let currentVerse = 1; currentVerse <= Number(lastVerseNum); currentVerse++) {
        passageBlock += `${currentVerse} ${bible[book][currentChapter][currentVerse]} `;
      }
    } else {
      passageBlock += '<br />';
      for (let currentVerse = 1; bible[book][currentChapter][currentVerse] !== undefined; currentVerse++) {
        passageBlock += `${currentVerse} ${bible[book][currentChapter][currentVerse]} `;
      }
      passageBlock += '<br />';
    }
  }
```

File: ui/src/components/helpers/components/BibleVerseAccordion.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ export const BibleVerseAccordion = ({ bibleVerse, actionText }) => {
           </Box>
           <AccordionIcon />
         </AccordionButton>
-        <AccordionPanel pb={4}>{passage}</AccordionPanel>
+        <AccordionPanel pb={4} dangerouslySetInnerHTML={{ __html: passage }} />
```
Comment (bhryck) [inline ui/src/components/helpers/components/BibleVerseAccordion.js:29]: why are we using this ya

**Review by bhryck — APPROVED** [review]

File: ui/src/components/helpers/components/BibleVerseAccordion.js (hunk 3 above)
Comment (Midnite1212) [reply ui/src/components/helpers/components/BibleVerseAccordion.js:29]: To resolve the line breaks, cuz tiptap editable component doesnt handle line breaks, so need to manually set the inner html

**Pull request merged**